docs/src/main/tut/TypedDatasetVsSparkDataset.md

```diff
@@ -131,8 +131,42 @@ And the compiler is our friend.
 fds.filter( fds('i) === 10 ).select( fds('x) )
 ```
 
+## Differences in Encoders
+
+Encoders in Spark's `Datasets` are partially type-safe. If you try to create a `Dataset` using  a type that is not 
+ a Scala `Product` then you get a compilation error:
+
+```tut:book
+class Bar(i: Int)
+```
+
+`Bar` is not a case class and not a Product, so the following correctly gives a compilation error in Spark:
+
+```tut:fail
+spark.createDataset(Seq(new Bar(1)))
+```
+
+However, this guard is not entirely type-safe. There is nothing preventing us from creating a case class for which 
```

Comment:
Actually tut doesn't like a bock that is marked as failed to start with a correct statement. I tried it but didn't work. Might look strange, but for now I am keeping it separate.